client/my-sites/customer-home/locations/primary/index.jsx

```diff
@@ -48,23 +55,29 @@ const cardComponents = {
 	[ NOTICE_CELEBRATE_SITE_SETUP_COMPLETE ]: CelebrateSiteSetupComplete,
 };
 
-const Primary = ( { cards } ) => {
+const Primary = ( { cards, renderCard } ) => {
 	if ( ! cards || ! cards.length ) {
 		return null;
 	}
 
-	return (
-		<>
-			{ cards.map(
-				( card, index ) =>
-					cardComponents[ card ] &&
-					React.createElement( cardComponents[ card ], {
-						key: index,
-						isIos: card === 'home-task-go-mobile-ios' ? true : null,
-					} )
-			) }
-		</>
-	);
+	return <>{ cards.map( renderCard ) }</>;
```

Comment:
I think this is causing the WSOD, Let's maybe back this out in favor of adding impressions in the base helper components.

It's also fine to also reduce the scope of this PR to just task impressions, as we're most interested in that funnel. The other cards are a little less interesting (and we can add this in follow ups).